pandas/core/internals.py

```diff
@@ -2468,7 +2541,8 @@ class DatetimeBlock(DatetimeLikeBlockMixin, Block):
     _can_hold_na = True
 
     def __init__(self, values, placement, fastpath=False, **kwargs):
-        if values.dtype != _NS_DTYPE:
+        if values.dtype != _NS_DTYPE and values.dtype.base != _NS_DTYPE:
+            # not datetime64 or datetime64tz
```

Comment:
This was so `NonConsolidatableMixin` could call `__init__`. I *think* it's harmless.